### ISSUE TITLE
GMwork is not sign definite 

### ISSUE THREAD

**Issue by gustavo-marques** [NOAA-GFDL/MOM6#965]
We have found that GMwork can take both positive and negative signs.  We believe it should be negative definite. The spots where it is positive tend to occur in the high latitudes, but we have also identified certain locations north of Scotland where it occurs occasionally.  Below is a snapshot from OM4_05, showing positive values at high latitudes:

![image](https://user-images.githubusercontent.com/11339137/62168585-26377600-b2e3-11e9-98ee-f03204409404.png)

**Comment by adcroft** [comment]
Is this using the Ferarri et al., 2010, stream function? It's never been clear to me whether we expect the conversion term (loss of APE) to be locally sign definite.

**Comment by gustavo-marques** [comment]
Yes, this is using the Ferarri et al., 2010 stream function.

**Comment by StephenGriffies** [comment]
@adcroft is correct.  

In Section 4.3 of Ferrari, Griffies, Nurser, and Vallis (2010), we discuss the effects of the scheme on potential energy.  The effects are sign definite only when integrating over the full depth.  However, at any point within the fluid column the sign can be positive or negative.  We see that property not as a failure for the streamfunction scheme.  Instead, it is a reasonable feature since eddies can locally increase PE just as well as decrease, yet in the depth integral we constrain them to decrease PE.  This property does contrast to the GM scheme, with GM a local sink of PE everywhere.   

So, can you sum the diagnostic to confirm that the depth integral the MOM6 implementation provides a sign-definite sink?

**Comment by gustavo-marques** [comment]
@StephenGriffies, thanks for the explanation. 

The snapshot above, taken from OM4_05, is already the depth integral (GMwork, CMOR equivalent is "tnkebto").

**Comment by StephenGriffies** [comment]
oops.  Then there is in fact a problem.  Thanks for identifying this.  

We need to look into the diagnostic to see what's up.  Have you poked around at all?

**Comment by gustavo-marques** [comment]
No, we have not. I propose we talk about this during the next MOM6 dev call.

**Comment by gustavo-marques** [comment]
Update:
I've run OM4_05 with `KHTH_USE_FGNV_STREAMFUNCTION = False`.  Still getting GMwork > 0, but now positive values occur mostly near the coastline (red values below, the maximum value is 7E-3). I've also confirmed that N^2 (CMOR obvfsq) is always >= 0 everywhere. I get very similar GMwork values when I set `EQN_OF_STATE = "linear"`. 

![image](https://user-images.githubusercontent.com/11339137/62740896-4f988600-b9f6-11e9-8cc9-24e1d0529772.png)

**Comment by StephenGriffies** [comment]
Thanks @gustavo-marques for these tests.  They help much.  

The linear EOS test says that something must be wrong with the mimetic nature of the code, as the continuum GM and streamfunctions both show sign-definite sink of PE.

**Comment by gustavo-marques** [comment]
Update:
Setting USE_STORED_SLOPES = False does not solve the problem.

**Comment by StephenGriffies** [comment]
Is this issue still existing?  Any bright ideas?  Or is this something that we can live with so not big priority?

**Comment by gustavo-marques** [comment]
I believe this issue still exists, although it would be good to confirm it using the latest main branch. My plate is pretty full right now, but I will try to check this in the next week or so.

**Comment by gustavo-marques** [comment]
I run OM4_05 with the latest dev/gfdl for MOM6 and SIS2 and I still see positive values for GMwork. I definitely see improvements when compared to the first plot I shared in this post (see also comments in https://github.com/NOAA-GFDL/MOM6/pull/1050). For example, the patch of positive values in the Arctic Ocean is no longer there. There are only very few points where GMwork > 0 now, making it harder to visualize them (that's why I am not uploading a new plot).

**Comment by StephenGriffies** [comment]
Interesting.  Do you know what code change made the issue become less egregious?

**Comment by gustavo-marques** [comment]
Good question. I was about to say [this](https://github.com/NOAA-GFDL/MOM6/pull/1050/files#diff-f667ab3399aff360a13eefb0797f44b5489cca3f149cfcf41936579b20d87b95R1240-R1248) was the code change that made the difference. However, setting `USE_GM_WORK_BUG = True` results in very little difference, so I don't think this patch was the main reason for the improvement.

**Comment by gustavo-marques** [comment]
@marshallward, shouldn't [this line](https://github.com/NOAA-GFDL/MOM6/blob/dev/gfdl/src/parameterizations/lateral/MOM_thickness_diffuse.F90#L1066) be changed to `Work_u(I,j) = Work_u(I,j) - G_scale`, just like you did in https://github.com/NOAA-GFDL/MOM6/pull/1050 ? 
Similarly for Work_v [here](https://github.com/NOAA-GFDL/MOM6/blob/dev/gfdl/src/parameterizations/lateral/MOM_thickness_diffuse.F90#L1331).

**Comment by marshallward** [comment]
Possibly, I don't know the origin of that term .  The change in #1050 was to fix the inconsistency of the sign in the layer-1 correction for `Work_u` and `Work_v` as detected by the rotation.  @Hallberg-NOAA was the one who identified which term in #1050 was correct.

I guess all I am saying is that if one is wrong, then they're both wrong, but the rotation test would not have caught it.  I think someone tuned into the expression needs to comment.
